conda_smithy/configure_feedstock.py

```diff
@@ -177,420 +306,192 @@ def render_circle(jinja_env, forge_config, forge_dir):
+                            if line.strip() and not line.strip().startswith('#')]
+        if not requirements:
+            raise ValueError("No yum requirements enabled in the "
+                             "yum_requirements.txt, please remove the file "
+                             "or add some.")
+        build_setup += textwrap.dedent("""\
+
+            # Install the yum requirements defined canonically in the
+            # "recipe/yum_requirements.txt" file. After updating that file,
+            # run "conda smithy rerender" and this line be updated
+            # automatically.
+            /usr/bin/sudo -n yum install -y {}
+
+
+        """.format(' '.join(requirements)))
+
+    forge_config['build_setup'] = build_setup
+
+    # TODO: Conda has a convenience for accessing nested yaml content.
+    template = jinja_env.get_template('run_docker_build.tmpl')
+    target_fname = os.path.join(forge_dir, '.circleci', 'run_docker_build.sh')
+    with write_file(target_fname) as fh:
+        fh.write(template.render(**forge_config))
 
-        build_setup = ""
+    template_name = 'fast_finish_ci_pr_build.sh.tmpl'
+    template = jinja_env.get_template(template_name)
+    target_fname = os.path.join(forge_dir, '.circleci', 'fast_finish_ci_pr_build.sh')
+    with write_file(target_fname) as fh:
+        fh.write(template.render(**forge_config))
 
-        # If the recipe supplies its own conda-forge-build-setup script,
-        # we use it instead of the global one.
-        cfbs_fpath = os.path.join(forge_dir, 'recipe',
-                                  'run_conda_forge_build_setup_osx')
-        if os.path.exists(cfbs_fpath):
-            build_setup += textwrap.dedent("""\
-                # Overriding global conda-forge-build-setup with local copy.
-                source {recipe_dir}/run_conda_forge_build_setup_osx
-            """.format(recipe_dir=forge_config["recipe_dir"]))
-        else:
-            build_setup += textwrap.dedent("""\
-                source run_conda_forge_build_setup
-            """)
+    # Fix permissions.
+    target_fnames = [
+        os.path.join(forge_dir, '.circleci', 'checkout_merge_commit.sh'),
+        os.path.join(forge_dir, '.circleci', 'fast_finish_ci_pr_build.sh'),
+        os.path.join(forge_dir, '.circleci', 'run_docker_build.sh'),
+    ]
+    for each_target_fname in target_fnames:
+        set_exe_file(each_target_fname, True)
 
-        build_setup = build_setup.strip()
-        build_setup = build_setup.replace("\n", "\n      ")
 
-        forge_config['build_setup'] = build_setup
+def render_circle(jinja_env, forge_config, forge_dir):
+    target_path = os.path.join(forge_dir, '.circleci', 'config.yml')
+    template_filename = 'circle.yml.tmpl'
+    fast_finish_text = textwrap.dedent("""\
+            {get_fast_finish_script} | \\
+                 python - -v --ci "circle" "${{CIRCLE_PROJECT_USERNAME}}/${{CIRCLE_PROJECT_REPONAME}}" "${{CIRCLE_BUILD_NUM}}" "${{CIRCLE_PR_NUMBER}}"  # NOQA
```

Comment:
Should the `NOQA` be on the next line? Just noticed this got added to the CI files, which I'm guessing is not the intent. :)

Reply:
Moved down to next line.